pkgs/applications/window-managers/xborders/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, python3Packages, fetchFromGitHub, libwnck, gtk3, libnotify, wrapGAppsHook, gobject-introspection }:
+
+with python3Packages; buildPythonPackage rec {
+  pname = "xborders";
+  version = "3.4"; # in version.txt
+  src = fetchFromGitHub {
+    owner = "deter0";
+    repo = "xborder";
+    rev = "d840ed649e92b7ad399cdb1a30aeca7d09f2c08f";
+    sha256 = "sha256-Q2WWXekUL/OOr/OBqJOZbAHaWURbP+coBVp206uPSNY=";
+  };
+  buildInputs = [ libwnck gtk3 libnotify ];
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection ];
+  propagatedBuildInputs = [ pycairo requests pygobject3 ];
+  preBuild = ''
+    sed '52,58!d' ${pname}
+    sed -i '51 a\    return ${version}' ${pname}
+    cat >setup.py <<'EOF'
+    from setuptools import setup
+    setup(
+      name='${pname}',
+      version='${version}',
+      author='deter0',
+      description='${meta.description}',
+      install_requires=['pycairo', 'requests', 'PyGObject'],
+      scripts=[
+        '${pname}',
+      ],
+    )
+    EOF
+  '';
+  meta = with lib; {
+    description = "Active window border replacement for window managers.";
+    homepage = "https://github.com/deter0/xborder";
+    license = licenses.unlicense;
+    maintainers = with maintainers; [ elnudev ];
+    platforms = platforms.linux;
+  };
```

Comment:
```suggestion
  meta = {
    description = "Active window border replacement for window managers.";
    homepage = "https://github.com/deter0/xborder";
    license = lib.licenses.unlicense;
    maintainers = with lib.maintainers; [ elnudev ];
    platforms = lib.platforms.linux;
  };
```

Reply:
Is this necessary? In my previous packaging PRs this wasn't an issue, it seems overly verbose without the `with lib;`. I think most of the package derivations in nixpkgs I've looked at so far do it this way.

Reply:
> Is this necessary? 

No. The code will not melt the computer if you don't do this.

On the other hand, there are some arguments against `with` itself as a bad language construction. The main arguments are compiled [here](https://nix.dev/anti-patterns/language#with-attrset-expression).

The text above is too radical for some, but the nested `with` is worth noticing:

https://github.com/NixOS/nix/issues/490

https://github.com/pikajude/nixpkgs/commit/eea78e554e03906524f87406bb3a17d5c5264029

> In my previous packaging PRs this wasn't an issue

Yes. It happens to everyone.

Me included - some old expressions I wrote contain much of it, because, let's be honest, almost everyone copy-pastes those blocks.

> it seems overly verbose without the `with lib;`.

"Excessive verbosity" was never a sufficient reason to start every derivation with

```nix
{ pkgs }: with pkgs;
mkDerivation { . . . }
```

> I think most of the package derivations in nixpkgs I've looked at so far do it this way.

Yes, and it should be changed.

Reply:
I understand the issue with having a `with` at the beginning of the derivation, but I'm still a bit confused what the problem is with using `with lib;` in `meta`. It's only a small set and almost all of the properties' values are inside `lib`, so it doesn't seem to me like there would be any downsides of using it there. What's the advantage of omitting it?

I'll make the changes.

Reply:
If the example in libuv quoted above - and below, again

https://github.com/pikajude/nixpkgs/commit/eea78e554e03906524f87406bb3a17d5c5264029

is not convincing, and you believe there is absolutely nothing wrong in code like

```nix
   platforms = with platforms; linux ++ darwin;
```

then I have no more educational devices to rely on. Copy-paste as you wish.